column_test.go

```diff
@@ -397,3 +410,114 @@ func (i *fileOffsetIndex) CompressedPageSize(j int) int64 {
 	return int64(i.PageLocations[j].CompressedPageSize)
 }
 func (i *fileOffsetIndex) FirstRowIndex(j int) int64 { return i.PageLocations[j].FirstRowIndex }
+
+type columnStats struct {
+	page       int
+	columnType parquet.Type
+	minValues  []parquet.Value
+	maxValues  []parquet.Value
+}
+
+func newColumnStats(columnType parquet.Type) *columnStats {
+	return &columnStats{columnType: columnType}
+}
+
+func (c *columnStats) observe(value parquet.Value) {
+	if c.page >= len(c.minValues) {
+		c.minValues = append(c.minValues, value)
+	} else if c.columnType.Compare(c.minValues[c.page], value) > 0 {
+		c.minValues[c.page] = value
+	}
+
+	if c.page >= len(c.maxValues) {
+		c.maxValues = append(c.maxValues, value)
+	} else if c.columnType.Compare(c.maxValues[c.page], value) < 0 {
+		c.maxValues[c.page] = value
+	}
+}
+
+func (c *columnStats) pageRead() {
+	c.page++
+}
+
+type indexOrder int
+
+const (
```

Comment:
Ah, is it this??

Reply:
Basically, yes. This is a const representation of the expected values.